libs/zephyr-agent/src/lib/build-context/git-provider-utils.ts

```diff
@@ -0,0 +1,102 @@
+import gitUrlParse from 'git-url-parse';
+
+// Standard Git provider domains mapping
+const STANDARD_DOMAINS: Record<string, string> = {
+  'github.com': 'github',
+  'gitlab.com': 'gitlab',
+  'bitbucket.org': 'bitbucket',
+};
+
+const STANDARD_DOMAIN_NAMES = Object.keys(STANDARD_DOMAINS);
+
+/**
+ * Git provider detection and information extraction. In Zephyr, application_uid is
+ * created as: [app_name, git_repo, git_org].join('.') where app_name comes from
+ * package.json name field, not from the git URL.
+ */
+export function getGitProviderInfo(gitUrl: string): {
+  provider: string;
+  owner: string;
+  project: string;
+  isEnterprise: boolean;
+} {
+  if (!gitUrl) {
+    throw new Error('Git URL is required');
+  }
+
+  const parsed = gitUrlParse(gitUrl);
+
+  // Detect if this is an enterprise/self-hosted instance
+  const isEnterprise = !STANDARD_DOMAIN_NAMES.includes(parsed.resource);
```

Comment:
does exactly the same as `determineProvider` and `STANDARD_DOMAIN_NAMES` is not needed if you remove duplication of logic

Reply:
Added and cleaned up